projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/sample/SampleRegistrationRequest.java

```diff
@@ -15,21 +16,32 @@
  * @param experimentalGroupId the experimental group id the sample is part of
  * @param replicateReference  the biological replicated reference the sample has been taken from
  * @param sampleOrigin        information about the sample origin.
+ * @param analysisType        analysis to be performed
+ * @param comment             comment relating to the sample
  * @since 1.0.0
  */
 public record SampleRegistrationRequest(String label, BatchId assignedBatch,
                                         ExperimentId experimentId, Long experimentalGroupId,
                                         BiologicalReplicateId replicateReference,
-                                        SampleOrigin sampleOrigin) {
+                                        SampleOrigin sampleOrigin, String analysisType, String comment) {
 
   public SampleRegistrationRequest(String label, BatchId assignedBatch, ExperimentId experimentId,
       Long experimentalGroupId, BiologicalReplicateId replicateReference,
-      SampleOrigin sampleOrigin) {
+      SampleOrigin sampleOrigin, String analysisType, String comment) {
     this.label = Objects.requireNonNull(label);
     this.assignedBatch = Objects.requireNonNull(assignedBatch);
     this.experimentId = Objects.requireNonNull(experimentId);
     this.experimentalGroupId = Objects.requireNonNull(experimentalGroupId);
     this.replicateReference = Objects.requireNonNull(replicateReference);
     this.sampleOrigin = Objects.requireNonNull(sampleOrigin);
+    this.comment = comment;
+    this.analysisType = analysisType;
+  }
+
+  public Optional<String> getAnalysisType() {
+    return Optional.ofNullable(analysisType);
+  }
+  public Optional<String> getComment() {
+    return Optional.ofNullable(comment);
```

Comment:
I read up on it. As soon as you have optional fields you should not use a record. The getter will still give back `null` so if you call `sampleRegistrationRequest.analysisType()` you would get `null`.
We should convert this to a class instead.

Reply:
I had this problem when I used the original method names "analysisType()" and "comment()". The suggestion to fix this problem was to explicitly add the "get" to the method name. This difference is also displayed by the IDE:
<img width="688" alt="image" src="https://github.com/qbicsoftware/data-manager-app/assets/10232219/d9447ef4-c524-4bf9-bbef-88faf728b399">
See first reply here for the solution I used: https://stackoverflow.com/questions/62945049/java-records-with-nullable-components

Reply:
yes that happens because there would be two methods with the same signature and name but different return type
`public String analysisType()` and `public Optional<String> analysisType()` which is not possible in Java. That is why I recommend overwriting the method and making it a class instead of a record or not using optional in this case but simply returning null and make use of optional in the Sample class.